sway-lib-std/src/storage.sw

```diff
@@ -1,11 +1,34 @@
 library storage;
 
+/// Store a stack variable in storage.
 pub fn store<T>(key: b256, value: T) {
-    asm(r1: key, r2: value) {
-        sww r1 r2;
+    if !is_reference_type::<T>() {
+        // If copy type, then it's a single word and can be stored with a single SWW.
+        asm(r1: key, r2: value) {
+            sww r1 r2;
+        };
+    } else {
+        // If reference type, then it's more than a word. Loop over every 32 bytes and
+        // store sequentially.
+        let mut size_left = size_of::<T>();
+        let mut ptr = asm(r1, r2: value) {
```

Comment:
@adlerjohn In the case of a reference type (which this is), isn't `value` already a pointer?

Reply:
Maybe. The code I wrote is completely untested!

Reply:
OK, this is a cast though, from `*b256` to `u64`.  Yet again, we should have an intrinsic for `SWW` et al. to avoid this one day.

Reply:
Ugh, I want to unresolve it to make a point.